### PR TITLE
Detectors that fail verification should still report the unverified secret

### PR DESCRIPTION
**Pull request by trufflesteeeve**
A number of detectors would fail to detect a secret if verification is turned on, but the verification request failed. This adjusts them to ensure the secret is still included if the verification request fails.

I do have a couple of questions that are slightly unrelated. I've included comments in the code for them, and will add PR comments on them as well.

### CODE REVIEW

File: pkg/detectors/github_old/github_old.go [hunk 1]
```diff
@@ -75,21 +75,24 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
 
+		if !s.Verified && detectors.IsKnownFalsePositive(token, detectors.DefaultFalsePositives, true) {
+			continue
+		}
+
 		results = append(results, s)
 	}
 
-	return
+	return // Should we add detectors.CleanResults(results), nil ?
```
Comment (trufflesteeeve) [inline pkg/detectors/github_old/github_old.go:97]: This detector looks like it might have a lot of false positives. Should the results be cleaned like similar detectors?

File: pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ var (
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
 func (s Scanner) Keywords() []string {
-	return []string{"microsoft"}
+	return []string{"microsoft"} // should we instead look for webhook.office.com, or something required to be in the regex above?
```
Comment (trufflesteeeve) [inline pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go:30]: This is unrelated to the overall goal, so I'm happy to pull it into another PR. But I think if the key must have `webhook.office.com` in the key pattern, that's probably fine to use as the keyword as well.

File: pkg/detectors/onelogin/onelogin.go [hunk 3]
```diff
@@ -63,26 +63,23 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
+			if !s.Verified && detectors.IsKnownFalsePositive(string(s.Raw), detectors.DefaultFalsePositives, true) {
+				continue
 			}
 
 			results = append(results, s)
 		}
 	}
 
-	return
+	return detectors.CleanResults(results), nil
```
Comment (trufflesteeeve) [inline pkg/detectors/onelogin/onelogin.go:84]: I wasn't sure if this would have a number of false positives. So I could see this line being removed.

File: pkg/detectors/pivotaltracker/pivotaltracker.go [hunk 4]
```diff
@@ -57,24 +56,21 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
-			}
+		if !s.Verified && detectors.IsKnownFalsePositive(string(s.Raw), detectors.DefaultFalsePositives, true) {
+			continue
 		}
 
 		results = append(results, s)
 	}
 
-	return
+	return detectors.CleanResults(results), nil
```
Comment (trufflesteeeve) [inline pkg/detectors/pivotaltracker/pivotaltracker.go:75]: Again, trying to not get lots of false positives.

File: pkg/detectors/rechargepayments/rechargepayments.go [hunk 5]
```diff
@@ -54,25 +54,22 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
+			if !s.Verified && detectors.IsKnownFalsePositive(string(s.Raw), detectors.DefaultFalsePositives, true) {
+				continue
 			}
 
 			results = append(results, s)
 		}
 	}
 
-	return
+	return detectors.CleanResults(results), nil
```
Comment (trufflesteeeve) [inline pkg/detectors/rechargepayments/rechargepayments.go:74]: Another false positive check. Should it be here?

File: pkg/detectors/uri/uri.go [hunk 6]
```diff
@@ -109,34 +108,25 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
-			}
+		if !s.Verified && detectors.IsKnownFalsePositive(string(s.Raw), detectors.DefaultFalsePositives, false) {
+			continue
 		}
 
 		results = append(results, s)
 	}
 
-	return
+	return detectors.CleanResults(results), nil
```
Comment (trufflesteeeve) [inline pkg/detectors/uri/uri.go:131]: I should have started a review earlier, but yeah, another false positive check. Does it make sense?

File: pkg/detectors/github_old/github_old.go (hunk 1 above)
Comment (dustin-decker) [reply pkg/detectors/github_old/github_old.go:97]: Yeah I think so. The old tokens were quite noisy (commit hashes are detected)

File: pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go (hunk 2 above)
Comment (dustin-decker) [reply pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go:30]: Yes, that should replace the existing keyword. It is ideal to use something that must be in the pattern if possible.

File: pkg/detectors/jdbc/jdbc.go [hunk 7]
```diff
@@ -45,29 +45,26 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
+		//		continue
+		//	}
+		//	req.Header.Add("Accept", "application/json;")
+		//	req.Header.Add("Jdbc-Token", token)
+		//	res, err := client.Do(req)
+		//	if err == nil {
+		//		if res.StatusCode >= 200 && res.StatusCode < 300 {
+		//			s.Verified = true
+		//		}
+		//	}
```
Comment (dustin-decker) [inline pkg/detectors/jdbc/jdbc.go:63]: Can you delete all of this? No way that we'll try to verify with this. We have ticket tracking this one. It'll attempt to connect directly from the detector.

File: pkg/detectors/squareapp/squareapp.go [hunk 8]
```diff
@@ -75,21 +74,18 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
-					return results, err
-				}
-				defer res.Body.Close()
+				if err == nil {
+					res.Body.Close() // The request body is unused.
 
-				// 404 = Correct crentials. The fake access token should not be found
-				if res.StatusCode == http.StatusNotFound {
-					s.Verified = true
+					// 404 = Correct crentials. The fake access token should not be found
```
Comment (dustin-decker) [inline pkg/detectors/squareapp/squareapp.go:80]: typo `crentials`

**Review by dustin-decker — COMMENTED** [review]
LGTM except for the 3 requests that I commented on (keyword change, clean up commented code, and typo)

**Review by dustin-decker — APPROVED** [review]
